### PR TITLE
Implement heap sort in C#

### LINKED ISSUE #625: Heap Sort in C#
This is an open issue for the Heap Sort algorithm implementation in C#. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by MarkMuth**
**Fixes #625**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-hsrt/csharp/HeapSort.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            SwapValues(root, index, heap);
+
+            heapSize = index;
+            BuildMaxHeap(root, heapSize, heap);
+        }
+    }
+    
+    public static void Main()
+    {
+        int[] arr = new int[] {2, 3, 0, 4};
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/csharp/HeapSort.cs:84]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357874)

File: algorithms/ar-hsrt/csharp/HeapSort.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        DoHeapSort(arr);
+        foreach(int element in arr)
+        {
+            Console.Write(element + " ");
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/csharp/HeapSort.cs:88]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357874)

File: algorithms/ar-hsrt/csharp/HeapSort.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        Console.WriteLine("");
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/csharp/HeapSort.cs:90]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357874)

File: algorithms/ar-hsrt/csharp/HeapSort.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
+using System;
+
+public class HeapSort
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/csharp/HeapSort.cs:3]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357874)

File: algorithms/ar-hsrt/csharp/HeapSort.cs (hunk 1 above)
Comment (MarkMuth) [reply algorithms/ar-hsrt/csharp/HeapSort.cs:84]: Well, this part is just copy-paste from the template, but could of course be replaced by var.

File: algorithms/ar-hsrt/csharp/HeapSort.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            SwapValues(root, index, heap);
+
+            heapSize = index;
+            BuildMaxHeap(root, heapSize, heap);
+        }
+    }
+
+    public static void Main()
+    {
+        var arr = new int[] {2, 3, 0, 4};
```
Comment (RavinduSachintha) [inline algorithms/ar-hsrt/csharp/HeapSort.cs:84]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6357874)

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @MarkMuth 
Can you update the main Readme.md file with the algorithm you implemented?
Just update the file and send the commit to this PR, 
thanks

**Comment by MarkMuth** [comment]
Heap sort in C# is now listed in the README. Thanks for the hint.

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@all-contributors please add @MarkMuth for code and doc

**Comment by RavinduSachintha** [comment]
@MarkMuth thank you for your contribution. keep it up :+1:  make sure to star this repo :wink:

**Pull request merged**